docs/hugging_face.md

```diff
@@ -0,0 +1,82 @@
+
+```py
+from sentence_transformers import SentenceTransformer
+model = SentenceTransformer('model_name')
+```
+
+You can even click `Use in sentence-transformers` to get a code snippet that you can copy and paste! 
+
+<div style="display:flex; flex-direction:column; gap: 15px; margin-bottom: 15px;">
+<img style=max-height:150px;object-fit:contain;" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/libraries-sentence_transformers_snippet1.png"/>
+<img style="max-height:130px;object-fit:contain" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/libraries-sentence_transformers_snippet2.png"/>
+</div>
+
+Here is an example that loads the [multi-qa-MiniLM-L6-cos-v1 model](https://huggingface.co/sentence-transformers/multi-qa-MiniLM-L6-cos-v1) and uses it to encode sentences and then compute the distance between them for doing semantic search.
+
+```py
+from sentence_transformers import SentenceTransformer, util
+model = SentenceTransformer('multi-qa-MiniLM-L6-cos-v1')
+
+query_embedding = model.encode('How big is London')
+passage_embedding = model.encode(['London has 9,787,426 inhabitants at the 2011 census',
+                                  'London is known for its finacial district'])
+
+print("Similarity:", util.dot_score(query_embedding, passage_embedding))
+```
+
+Here is another example, this time using the [clips/mfaq model](https://huggingface.co/clips/mfaq) for multilingual FAQ retrieval. After embedding the query and the answers, we perform a semantic search to find the most relevant answer. 
+
+```py
+from sentence_transformers import SentenceTransformer, util
+
+question = "<Q>How many models can I host on HuggingFace?"
+answer_1 = "<A>All plans come with unlimited private models and datasets."
+answer_2 = "<A>AutoNLP is an automatic way to train and deploy state-of-the-art NLP models, seamlessly integrated with the Hugging Face ecosystem."
+answer_3 = "<A>Based on how much training data and model variants are created, we send you a compute cost and payment link - as low as $10 per job."
+
+model = SentenceTransformer('clips/mfaq')
+query_embedding = model.encode(question)
+corpus_embeddings = model.encode([answer_1, answer_2, answer_3])
+
+print(util.semantic_search(query_embedding, corpus_embeddings))
+```
+
+## Sharing your models
+
+You can share your SentenceTransformers models by using the [`save_to_hub` method](https://www.sbert.net/docs/package_reference/SentenceTransformer.html#sentence_transformers.SentenceTransformer.save_to_hub) from a trained model.
+
+```py
+from sentence_transformers import SentenceTransformer
+
+# Load or train a model
+model.save_to_hub("my_new_model")
+```
+
+This command creates a [Hugging Face repository](https://huggingface.co/docs/hub/repositories) with an automatically generated model card, an inference widget, example code snippets, and more! [Here](https://huggingface.co/osanseviero/my_new_model) is an example.
```

Comment:
WDYT of adding information of the new argument, `train_datasets`, for adding train datasets to the hub? Also, I propose we use [this model](https://huggingface.co/ITESM/sentece-embeddings-BETO) as an example since it already has "Dataset used to train".

```suggestion
This command creates the "namespace/my_new_model" [Hugging Face repository](https://huggingface.co/docs/hub/repositories). The namespace can be an individual account or an organization you have write access to (for example, 'fastai/sentence-transformer'). `save_to_hub` automatically generates a model card, an inference widget, example code snippets, and more. With the argument `train_datasets: Optional[List[str]] = None)` you can automatically add to the Hub's model carl a list of datasets you used to train the model; e.g., model.save_to_hub("namespace/my_new_model", train_datasets=["GEM/wiki_lingua", "code_search_net"]. [Here](https://huggingface.co/ITESM/sentece-embeddings-BETO) is share model example.
```

Reply:
Good idea! I added it in with some modifications